icon/iiss/base.go

```diff
@@ -178,8 +179,42 @@ func handleConsensusInfo(wc state.WorldContext) error {
 		return err
 	}
 
-	// TODO update P-Rep status
+	// update P-rep status, vtotal, vfail, vfailcont
+	proposerExist := false
+	for _, p := range prepAddressList {
+		if p == proposer {
+			proposerExist = true
+		}
+	}
+	if !proposerExist {
+		prepAddressList = append(prepAddressList, proposer)
+	}
+	err := handlePrepStatus(es.State, prepAddressList, voted)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func handlePrepStatus(state *icstate.State, prepAddressList []module.Address, voted []bool) error {
+	if len(prepAddressList) != 0 {
```

Comment:
You can delete this if statement for optimization

Reply:
If you mean that combine two methods into one, I intended to seperate those to make base_test.go

Reply:
No. I talked about 'if' statement.
```go
if len(prepAddressList) != 0 {
		for i := 0; i < len(prepAddressList); i += 1 {
```
If prepAddressList has no value, for loop will not do anything.

Therefore, 'if' statement is meaningless

Reply:
understood, I reflected it, thank you